controllers/backend/publicdomain/publicdomain.go

```diff
@@ -0,0 +1,120 @@
+package publicdomain
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rio/pkg/apply"
+	"github.com/rancher/rio/pkg/certs"
+	"github.com/rancher/rio/pkg/deploy/stack/populate/istio"
+	"github.com/rancher/rio/pkg/namespace"
+	"github.com/rancher/rio/pkg/settings"
+	"github.com/rancher/rio/types"
+	"github.com/rancher/rio/types/apis/rio.cattle.io/v1beta1"
+	spacev1beta1 "github.com/rancher/rio/types/apis/space.cattle.io/v1beta1"
+	corev1 "github.com/rancher/types/apis/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func Register(ctx context.Context, rContext *types.Context) {
+	dc := &domainController{
+		namespacesLister: rContext.Core.Namespaces("").Controller().Lister(),
+		stackLister:      rContext.Rio.Stacks("").Controller().Lister(),
+		serviceLister:    rContext.Rio.Services("").Controller().Lister(),
+		services:         rContext.Rio,
+		domains:          rContext.Global.PublicDomains(settings.RioSystemNamespace),
+		domainLister:     rContext.Global.PublicDomains(settings.RioSystemNamespace).Controller().Lister(),
+		routesetLister:   rContext.Rio.RouteSets("").Controller().Lister(),
+		routesets:        rContext.Rio,
+	}
+	rContext.Global.PublicDomains(settings.RioSystemNamespace).AddLifecycle(ctx, "public-domain-controller", dc)
+}
+
+type domainController struct {
+	namespacesLister corev1.NamespaceLister
+	stackLister      v1beta1.StackLister
+	serviceLister    v1beta1.ServiceLister
+	services         v1beta1.ServicesGetter
+	domains          spacev1beta1.PublicDomainInterface
+	domainLister     spacev1beta1.PublicDomainLister
+	routesetLister   v1beta1.RouteSetLister
+	routesets        v1beta1.RouteSetsGetter
+}
+
+func (d *domainController) Create(domain *spacev1beta1.PublicDomain) (runtime.Object, error) {
+	return domain, nil
+}
+
+func (d *domainController) Updated(domain *spacev1beta1.PublicDomain) (runtime.Object, error) {
+	if err := apply.Apply([]runtime.Object{certs.AcmeIssuer()}, nil, "", "acme-cluster-issuer"); err != nil {
+		return domain, err
+	}
+	ns, err := d.getNamespace(domain)
+	if err != nil {
+		return domain, err
+	}
+	// certificate
+	cert := certs.CertificateHttp(domain)
+	if err := apply.Apply([]runtime.Object{cert}, nil, settings.RioSystemNamespace, "certificate"); err != nil {
```

Comment:
I'm thinking you want a different groupID.  The groupID determines the set of object to watch for modification and prune.  In general the groupID is a unique for the input data.  What will happen with how this is at the moment is that domain foo.com will create `cert` and then bar.com will create `newCert` and prune (delete) `cert`